packages/edit-site/src/components/post-list/index.js

```diff
@@ -55,14 +67,15 @@ function useView( postType ) {
 			};
 		}
 		return defaultView;
-	}, [ isCustom, activeView, layout, postType, DEFAULT_VIEWS ] );
-	const [ view, setView ] = useState( selectedDefaultView );
-
+	}, [ activeView ] );
```

Comment:
Unless I'm missing some use case, we shouldn't care about any URL change other than `activeView`. The goal here is to replace the view completely when the `activeView` changes (the effect below), but ignore any other change.

Reply:
I'm not sure that's correct, what happens if there's an action that programmatically changes the url or things like that.

Reply:
I'm not sure I follow. Shouldn't we only reset the full view when the `activeView` changes? That's what is telling us to "load a different default view". How does it matter if it's updated manually or automatically?

Reply:
What I'm saying is that if the "layout" change in the url from another place than the current component, the current component should adapt. (For instance a command palette that switches a layout).  Your comment seem to suggest that it won't.

Reply:
Ah, I see your point now. So, we have two things to care about:

- reset the view when the user selects one of the default views
- update the view when layout is updated programatically

I pushed https://github.com/WordPress/gutenberg/pull/63889/commits/4bf53ef0b60eee7d1b73cab1c7ac68b5425e259e to address the second concern. It's a separate effect that selectively updates the view (only `view.type` changes, we don't reset any other thing).

I also realized that we still need to reset the view completely in the case `postType` or `isCustom` change programatically, so I brought back those as well.

Reply:
Hi @oandregal, We also need to take into account the layout URL change. The trash view uses layout table instead of list.